service/src/main/java/greencity/service/CommitInfoServiceImpl.java

```diff
@@ -0,0 +1,55 @@
+package greencity.service;
+
+import greencity.constant.AppConstant;
+import greencity.dto.commitinfo.CommitInfoDto;
+import greencity.dto.commitinfo.CommitInfoErrorDto;
+import greencity.dto.commitinfo.CommitInfoSuccessDto;
+import jakarta.annotation.PostConstruct;
+import java.io.File;
+import java.io.IOException;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.revwalk.RevWalk;
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder;
+import org.springframework.stereotype.Service;
+
+/**
+ * Service implementation for fetching Git commit information.
+ */
+@Service
+public class CommitInfoServiceImpl implements CommitInfoService {
+    private Repository repository;
+
+    @PostConstruct
+    private void init() {
+        try {
+            repository = new FileRepositoryBuilder()
+                .setGitDir(new File(".git"))
+                .readEnvironment()
+                .findGitDir()
+                .build();
+        } catch (IOException e) {
+            throw new IllegalStateException("Failed to initialize repository", e);
```

Comment:
Do we need to fail the application load if the `.git` is not found?

Reply:
Ok, it will be handled